### ISSUE TITLE
Error: exit status 2, Failed running "start"

### ISSUE THREAD

**Issue by ansarizafar** [cockroachdb/cockroach#14443]
I am getting this error on Ubuntu 16.04

![cockroachdb](https://cloud.githubusercontent.com/assets/5788284/24458094/60ab5900-14b1-11e7-8167-c3f18d9bbf58.png)

**Comment by tamird** [comment]
Can you share your logs?

**Comment by ansarizafar** [comment]
Here is the log
````
zafar@home:~/Downloads/cockroach-beta-20170323.linux-amd64$ ./cockroach start —logtostderr
Usage:
  cockroach start [flags]

Examples:
  cockroach start --insecure --store=attrs=ssd,path=/mnt/ssd1 [--join=host:port,[host:port]]

Flags:
      --advertise-host string         
        The hostname to advertise to other CockroachDB nodes for intra-cluster
        communication; it must resolve from other nodes in the cluster.
       
      --attrs string                  
        An ordered, colon-separated list of node attributes. Attributes are
        arbitrary strings specifying machine capabilities. Machine capabilities
        might include specialized hardware or number of cores (e.g. "gpu",
        "x16c"). For example:

          --attrs=x16c:gpu
       
      --background                    
        Start the server in the background. This is similar to appending "&"
        to the command line, but when the server is started with --background,
        control is not returned to the shell until the server is ready to
        accept requests.
       
      --ca-cert string                
        Path to the CA certificate. Needed by clients and servers in secure
        mode.
        Environment variable: COCKROACH_CA_CERT
       
      --cache bytes                   
        Total size in bytes for caches, shared evenly if there are multiple
        storage devices. Size suffixes are supported (e.g. 1GB and 1GiB). If
        left unspecified, defaults to 25% of the physical memory, or 512MB if
        the memory size cannot be determined.
        (default 962 MiB)
      --cert string                   
        Path to the client or server certificate. Needed in secure mode.
        Environment variable: COCKROACH_CERT
       
      --host string                   
        The hostname to listen on. The node will also advertise itself using
        this hostname if advertise-host is not specified.
       
      --http-host string              
        The hostname or IP address to bind to for HTTP requests.
       
      --http-port string              
        The port to bind to for HTTP requests.
        (default "8080")
      --insecure                      
        Run over non-encrypted (non-TLS) connections. This is strongly
        discouraged for production usage and this flag must be explicitly
        specified in order for the server to listen on an external address in
        insecure mode.
        Environment variable: COCKROACH_INSECURE
        (default true)
  -j, --join string                   
        The address of node which acts as bootstrap when a new node is joining
        an existing cluster. This flag can be specified separately for each
        address, for example:

          --join=localhost:1234 --join=localhost:2345

        Or can be specified as a comma separated list in single flag, or both
        forms can be used together, for example:

          --join=localhost:1234,localhost:2345 --join=localhost:3456

        Each address in the list has an optional type: [type=]<address>. An
        unspecified type means ip address or dns. Type is one of:

          - tcp: (default if type is omitted): plain ip address or hostname.
          - http-lb: HTTP load balancer: we query
                     http(s)://<address>/_status/details/local
       
      --key string                    
        Path to the key protecting --cert. Needed in secure mode.
        Environment variable: COCKROACH_KEY
       
      --locality Locality             
        Not fully implemented.
        https://github.com/cockroachdb/cockroach/issues/4868 An ordered,
        comma-separated list of key-value pairs that describe the topography
        of the machine. Topography might include country, datacenter or rack
        designations. Data is automatically replicated to maximize diversities
        of each tier. The order of tiers is used to determine the priority of
        the diversity. The tiers and order must be the same on all nodes.  For
        example:

          --locality=country=us,region=us-west,datacenter=us-west-1b,rack=12
          --locality=country=ca,region=ca-east,datacenter=ca-east-2,rack=4

          --locality=planet=earth,province=manitoba,colo=secondary,power=3
       
      --max-sql-memory bytes          
        Total size in bytes available for use to store temporary data for SQL
        clients, including prepared queries and intermediate data rows during
        query execution. Size suffixes are supported (e.g. 1GB and 1GiB). If
        left unspecified, defaults to 25% of the physical memory, or 512MB if
        the memory size cannot be determined.
        (default 962 MiB)
      --pid-file string               
        After the CockroachDB node has started up successfully, it will write
        its process ID to the specified file.
       
  -p, --port string                   
        The port to bind to.
        (default "26257")
      --raft-tick-interval duration   
        The resolution of the Raft timer; other raft timeouts are defined in
        terms of multiples of this value.
        (default 200ms)
  -s, --store StoreSpec               
        The file path to a storage device. This flag must be specified
        separately for each storage device, for example:

          --store=/mnt/ssd01 --store=/mnt/ssd02 --store=/mnt/hda1

        For each store, the "attrs" and "size" fields can be used to specify
        device attributes and a maximum store size (see below). When one or
        both of these fields are set, the "path" field label must be used for
        the path to the storage device, for example:

          --store=path=/mnt/ssd01,attrs=ssd,size=20GiB

        In most cases, node-level attributes are preferable to store-level
        attributes. However, the "attrs" field can be used to match
        capabilities for storage of individual databases or tables. For
        example, an OLTP database would probably want to allocate space for
        its tables only on solid state devices, whereas append-only time
        series might prefer cheaper spinning drives. Typical attributes include
        whether the store is flash (ssd), spinny disk (hdd), or in-memory
        (mem), as well as speeds and other specs. Attributes can be arbitrary
        strings separated by colons, for example:

          --store=path=/mnt/hda1,attrs=hdd:7200rpm

        The store size in the "size" field is not a guaranteed maximum but is
        used when calculating free space for rebalancing purposes. The size can
        be specified either in a bytes-based unit or as a percentage of hard
        drive space, for example:

          --store=path=/mnt/ssd01,size=10000000000     -> 10000000000 bytes
          --store-path=/mnt/ssd01,size=20GB            -> 20000000000 bytes
          --store-path=/mnt/ssd01,size=20GiB           -> 21474836480 bytes
          --store-path=/mnt/ssd01,size=0.02TiB         -> 21474836480 bytes
          --store=path=/mnt/ssd01,size=20%             -> 20% of available space
          --store=path=/mnt/ssd01,size=0.2             -> 20% of available space
          --store=path=/mnt/ssd01,size=.2              -> 20% of available space

        For an in-memory store, the "type" and "size" fields are required,
        and the "path" field is forbidden. The "type" field must be set to
        "mem", and the "size" field must be set to the true maximum bytes or
        percentage of available memory that the store may consume, for example:

          --store=type=mem,size=20GiB
          --store=type=mem,size=90%

        Commas are forbidden in all values, since they are used to separate
        fields. Also, if you use equal signs in the file path to a store, you
        must use the "path" field label.
        (default --store=path=cockroach-data)

Global Flags:
      --alsologtostderr Severity[=INFO]   logs at or above this threshold go to stderr (default INFO)
      --log-backtrace-at traceLocation    when logging hits line file:N, emit a stack trace (default :0)
      --log-dir string                    if non-empty, write log files in this directory
      --logtostderr                       log to standard error instead of files
      --no-color                          disable standard error log colorization
      --verbosity level                   log level for V logs
      --vmodule moduleSpec                comma-separated list of pattern=N settings for file-filtered logging
Error: invalid arguments
Failed running "start"
zafar@home:~/Downloads/cockroach-beta-20170323.linux-amd64$ 
````

**Comment by bdarnell** [comment]
From gitter: the CPU on this machine is `model name    : Pentium(R) Dual-Core  CPU      E6300  @ 2.80GHz`, which was released in 2006 and supports SSE version 3. In `c-rocksdb/cgo_cflags.go` we set `-msse4.2` (SSE 4.2 is supported in intel CPUs released in 2009 and later).

We should probably provide a build tag that can be used to remove the `-msse4.2` flag so that building from source on an older CPU will produce a working binary.

**Comment by tamird** [comment]
It's pretty annoying that we have to modify c-rocksdb to support these build configuration changes, though I don't have any suggestions for an alternative arrangement.

Can we have someone new to the c-* repos take this one? Perhaps @benesch is interested.

**Comment by benesch** [comment]
We could instead manipulate the flags using `CGO_*` environment variables. Perhaps we should set `-march=native` by default, and override it as necessary when we compile release builds? That's the simplest way I can think of to fix the problem.

**Comment by bdarnell** [comment]
If that works (`-m` flags specified in `CGO_CPPFLAGS` take precedence over the ones in `cgo_flags.go`), then that makes sense to me.

**Comment by petermattis** [comment]
I think we should also try to give a better warning if we're running on a processor which doesn't support sse4.2. See https://github.com/cockroachdb/cockroach/pull/14466#issuecomment-290394894.

**Comment by hrstoyanov** [comment]
Hello,
I am using the snapshot bild from 3/31/2-17 and I am getting this very same issue in VMWare vps instance:
```
-bash-4.1$ cat /etc/*-release
Red Hat Enterprise Linux Server release 6.8 (Santiago)
-bash-4.1$ uname -a
Linux betapmwsv17 2.6.32-642.13.1.el6.x86_64 #1 SMP Wed Nov 23 16:03:01 EST 2016 x86_64 x86_64 x86_64 GNU/Linux
-bash-4.1$ ./cockroach version
Build Tag:    beta-20170330
Build Time:   2017/03/30 15:14:06
Distribution: CCL
Platform:     linux amd64
Go Version:   go1.8
C Compiler:   gcc 6.3.0
Build SHA-1:  f0f5783cc67f9cda16cedf1b88ca0691b0b65884
Build Type:   release
-bash-4.1$ ./cockroach start --background  --port=26259 --http-port=9870
Error: exit status 2
Failed running "start"
-bash-4.1$  ./cockroach start --logtostderr
I170331 10:37:43.948204 1 cli/start.go:330  CockroachDB CCL beta-20170330 (linux amd64, built 2017/03/30 15:14:06, go1.8)
I170331 10:37:44.049240 39 cli/start.go:367  starting cockroach node
W170331 10:37:44.052801 39 server/server.go:158  [n?] running in insecure mode, this is strongly discouraged. See --insecure.
W170331 10:37:44.060834 39 server/config.go:352  soft open file descriptor limit 4096 is under the recommended limit 15000; this may decrease performance
please see https://www.cockroachlabs.com/docs/recommended-production-settings.html for more details
I170331 10:37:44.061145 39 storage/engine/rocksdb.go:374  opening rocksdb instance at "cockroach-data"
SIGILL: illegal instruction
PC=0x17525f9 m=6 sigcode=2
signal arrived during cgo execution

goroutine 39 [syscall, locked to thread]:
non-Go function
        pc=0x17525f9
non-Go function
        pc=0x176cad6
runtime.cgocall(0x1517a00, 0xc4200a0cd0, 0xc4200a0d40)
        /usr/local/go/src/runtime/cgocall.go:131 +0xe2 fp=0xc4200a0c60 sp=0xc4200a0c20
github.com/cockroachdb/cockroach/pkg/storage/engine._Cfunc_DBOpen(0xc4200a91e0, 0xc4203c6850, 0xe, 0x7fe163213010, 0x8000, 0x0, 0x400000000, 0xf00, 0x0, 0x0)
        github.com/cockroachdb/cockroach/pkg/storage/engine/_obj/_cgo_gotypes.go:568 +0x5d fp=0xc4200a0cd0 sp=0xc4200a0c60
github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).open.func1(0xc4200a91e0, 0xc4203c6850, 0xe, 0x7fe163213010, 0x8000, 0x0, 0x400000000, 0xf00, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:399 +0x1a1 fp=0xc4200a0d78 sp=0xc4200a0cd0
github.com/cockroachdb/cockroach/pkg/storage/engine.(*RocksDB).open(0xc4200a91e0, 0xc4201284e0, 0x12)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:399 +0x37d fp=0xc4200a0ed0 sp=0xc4200a0d78
github.com/cockroachdb/cockroach/pkg/storage/engine.NewRocksDB(0x0, 0x0, 0x0, 0x1cc37cd, 0xe, 0x7fe163213000, 0x0, 0xf00, 0x0, 0x0, ...)
        /go/src/github.com/cockroachdb/cockroach/pkg/storage/engine/rocksdb.go:340 +0x203 fp=0xc4200a0f60 sp=0xc4200a0ed0
github.com/cockroachdb/cockroach/pkg/server.(*Config).CreateEngines(0xc420094008, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/config.go:477 +0x982 fp=0xc4200a1268 sp=0xc4200a0f60
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start(0xc420094000, 0x7fe16709ebc0, 0xc4203c4510, 0xc420132540, 0x1)
        /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:606 +0xc22 fp=0xc4200a1a30 sp=0xc4200a1268
github.com/cockroachdb/cockroach/pkg/cli.runStart.func1.1(0x7fe16709ebc0, 0xc4202d2090, 0xc42000e048, 0xc4202a60a0, 0xc420364360, 0xc4202ac000, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:386 +0x203 fp=0xc4200a1f38 sp=0xc4200a1a30
github.com/cockroachdb/cockroach/pkg/cli.runStart.func1(0x294bac0, 0xc4202a6000, 0x7fe16709ebc0, 0xc4202d2090, 0xc42000e048, 0xc4202a60a0, 0xc420364360, 0xc4202ac000, 0xc4202a2420)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:440 +0xbb fp=0xc4200a1f98 sp=0xc4200a1f38
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2197 +0x1 fp=0xc4200a1fa0 sp=0xc4200a1f98
created by github.com/cockroachdb/cockroach/pkg/cli.runStart
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:443 +0x8ef

goroutine 1 [select]:
github.com/cockroachdb/cockroach/pkg/cli.runStart(0x291bec0, 0xc4203d56c0, 0x0, 0x1, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/start.go:452 +0x158a
github.com/cockroachdb/cockroach/pkg/cli.MaybeDecorateGRPCError.func1(0x291bec0, 0xc4203d56c0, 0x0, 0x1, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/error.go:32 +0x5a
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).execute(0x291bec0, 0xc4203d5690, 0x1, 0x1, 0x291bec0, 0xc4203d5690)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:644 +0x3f2
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x2917460, 0x6b6ffe, 0xc400000000, 0xc4202d5a70)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:734 +0x339
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).Execute(0x2917460, 0xc42009df00, 0xc4202d5a70)
        /go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:693 +0x2b
github.com/cockroachdb/cockroach/pkg/cli.Run(0xc420010220, 0x2, 0x2, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:122 +0x6d
github.com/cockroachdb/cockroach/pkg/cli.Main()
        /go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:45 +0xfb
main.main()
        /go/src/github.com/cockroachdb/cockroach/main.go:28 +0x20

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2197 +0x1

goroutine 5 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).flushDaemon(0x2d76740)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1004 +0x77
created by github.com/cockroachdb/cockroach/pkg/util/log.init.1
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:583 +0xd2

goroutine 6 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).gcDaemon(0x2d76740)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1027 +0x5b
created by github.com/cockroachdb/cockroach/pkg/util/log.init.1
        /go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:584 +0xf6

goroutine 7 [syscall]:
os/signal.signal_recv(0x0)
        /usr/local/go/src/runtime/sigqueue.go:116 +0x104
os/signal.loop()
        /usr/local/go/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.1
        /usr/local/go/src/os/signal/signal_unix.go:28 +0x41

goroutine 9 [select, locked to thread]:
runtime.gopark(0x1d37068, 0x0, 0x1cb9315, 0x6, 0x18, 0x2)
        /usr/local/go/src/runtime/proc.go:271 +0x13a
runtime.selectgoImpl(0xc42003bf50, 0x0, 0x18)
        /usr/local/go/src/runtime/select.go:423 +0x1364
runtime.selectgo(0xc42003bf50)
        /usr/local/go/src/runtime/select.go:238 +0x1c
runtime.ensureSigM.func1()
        /usr/local/go/src/runtime/signal_unix.go:434 +0x2dd
runtime.goexit()
        /usr/local/go/src/runtime/asm_amd64.s:2197 +0x1

goroutine 41 [chan receive]:
github.com/cockroachdb/cockroach/pkg/rpc.NewContext.func1()
        /go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:170 +0x76
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc4202a8460)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 42 [select]:
github.com/cockroachdb/cockroach/pkg/kv.(*TxnCoordSender).printStatsLoop(0xc4200a94a0, 0x7fe16709ebc0, 0xc4202d33e0)
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:207 +0xb4a
github.com/cockroachdb/cockroach/pkg/kv.NewTxnCoordSender.func1()
        /go/src/github.com/cockroachdb/cockroach/pkg/kv/txn_coord_sender.go:191 +0x98
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc420364d20)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 43 [select]:
github.com/cockroachdb/cockroach/pkg/sql.(*LeaseManager).RefreshLeases.func1()
        /go/src/github.com/cockroachdb/cockroach/pkg/sql/lease.go:1231 +0xb37
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc4202d4810)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 44 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/netutil.MakeServer.func2()
        /go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:98 +0x70
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc420128160)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 45 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/netutil.MakeServer.func2()
        /go/src/github.com/cockroachdb/cockroach/pkg/util/netutil/net.go:98 +0x70
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc4201281a0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 48 [chan receive]:
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func2()
        /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:491 +0x71
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc4203c4a50)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 49 [IO wait]:
net.runtime_pollWait(0x7fe1670a4368, 0x72, 0x0)
        /usr/local/go/src/runtime/netpoll.go:164 +0x59
net.(*pollDesc).wait(0xc420482148, 0x72, 0x0, 0xc4203c8000)
        /usr/local/go/src/net/fd_poll_runtime.go:75 +0x38
net.(*pollDesc).waitRead(0xc420482148, 0xffffffffffffffff, 0x0)
        /usr/local/go/src/net/fd_poll_runtime.go:80 +0x34
net.(*netFD).accept(0xc4204820e0, 0x0, 0x292eb20, 0xc4203c8000)
        /usr/local/go/src/net/fd_unix.go:430 +0x1e5
net.(*TCPListener).accept(0xc4201e6068, 0xc4202d2360, 0x1aeb520, 0x2903e10)
        /usr/local/go/src/net/tcpsock_posix.go:136 +0x2e
net.(*TCPListener).Accept(0xc4201e6068, 0xc4202d2270, 0x1aeb520, 0x2903e10, 0x1bca160)
        /usr/local/go/src/net/tcpsock.go:228 +0x49
net/http.(*Server).Serve(0xc4200a8b00, 0x293f0e0, 0xc4201e6068, 0x0, 0x0)
        /usr/local/go/src/net/http/server.go:2643 +0x228
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func5()
        /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:514 +0x42
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc4201283c0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 50 [chan receive]:
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func6()
        /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:518 +0x88
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc4203628c0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 51 [chan receive]:
github.com/cockroachdb/cockroach/vendor/github.com/cockroachdb/cmux.(*muxListener).Accept(0xc4201282e0, 0x1d363c8, 0xc4204f0000, 0x293fa60, 0xc4201282e0)
        <autogenerated>:9 +0x7a
github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).Serve(0xc4204f0000, 0x293fa60, 0xc4201282e0, 0x0, 0x0)
        /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:396 +0x189
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func7()
        /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:529 +0x43
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc4201283e0)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 52 [select]:
github.com/cockroachdb/cockroach/pkg/server.(*Server).Start.func9()
        /go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:577 +0x2da
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc420362900)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 53 [chan receive]:
github.com/cockroachdb/cockroach/pkg/gossip.(*server).start.func3()
        /go/src/github.com/cockroachdb/cockroach/pkg/gossip/server.go:366 +0x66
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc420128480)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 54 [select]:
github.com/cockroachdb/cockroach/pkg/gossip.(*Gossip).bootstrap.func1()
        /go/src/github.com/cockroachdb/cockroach/pkg/gossip/gossip.go:1037 +0x4da
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc4203c6760)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

goroutine 55 [select]:
github.com/cockroachdb/cockroach/pkg/gossip.(*Gossip).manage.func1()
        /go/src/github.com/cockroachdb/cockroach/pkg/gossip/gossip.go:1075 +0x399
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc4202a60a0, 0xc4203c6770)
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:210 +0x7d
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
        /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:211 +0x66

rax    0x7fe163215028
rbx    0x0
rcx    0x7fe163215038
rdx    0x0
rdi    0x7fe163ffa1c0
rsi    0x0
rbp    0x0
rsp    0x7fe163ffa0f0
r8     0x0
r9     0x7fe16328b830
r10    0x0
r11    0x206
r12    0xa
r13    0x7fe163215000
r14    0x7fe163ffa3b0
r15    0x0
rip    0x17525f9
rflags 0x10246
cs     0x33
fs     0x0
gs     0x0
```

**Comment by benesch** [comment]
@hrstoyanov, you could try building from source. If you manually remove "-msse4.2" from line 8 of `vendor/github.com/cockroachdb/c-rocksdb/cgo_flags.go` and compile, you should get a working binary. We're preparing a more permanent fix in #14466.

**Comment by edwardsmarkf** [comment]
any update on this?   a source install requires gcc6 and that may be an issue with centos7 .

**Comment by bdarnell** [comment]
You can build from source with our builder docker image: run `build/builder.sh make build` in the cockroach source directory. The resulting binaries will run on anything dating back to centos 6.

**Comment by edwardsmarkf** [comment]
well installing docker on centos7 sure is fun.....  but i get the same error as before:

**docker  run  --rm  cockroachdb/cockroach  start  --background  ;**
_Error: exist status 2
Failed running "start"_

any idea when this msse4.2 issue #14466 might be resolved?  i might just wait at this point, or try spanner.

**Comment by bdarnell** [comment]
Even when building from the docker image, you still need the change from https://github.com/cockroachdb/cockroach/issues/14443#issuecomment-290837169 (manually remove "-msse4.2" from line 8 of vendor/github.com/cockroachdb/c-rocksdb/cgo_flags.go) to support these old CPUs.

**Comment by edwardsmarkf** [comment]
sorry this is my first intro to docker (yum is so much easier)

at what point here would i intercept the process to remove the -msse4.2 part?  i tried to follow these [instructions](https://www.cockroachlabs.com/docs/install-cockroachdb.html).

docker run hello-world ;   ###  seems to work fine
docker pull cockroachdb/cockroach:latest  ;
docker run --rm  cockroachdb/cockroach  version ;
_Build Tag:    beta-20170330
Build Time:    2017/04/07 13:45:36
Distribution:   CCL
Platform:    linux amd64
Go Version:  go1.8
C Compiler:  gcc 6.3.0
Fuild SHA-1:   2039-blah-blah48203948320948
Build Type:     release_

**Comment by bdarnell** [comment]
You have to use the builder image, not the production docker image. Check out the cockroach source repo as if you're going to build from source, `git submodule update --init`, edit the file, then run `build/builder.sh make build`

**Comment by edwardsmarkf** [comment]
to be fair, this is a cento7 running on virtualbox.   the cpu is 
**AMD Phenom(tm) II X6 1100T Processor 3.30 Ghz**
however, it seems to run fine on centos7 google-cloud.

> ...to support these old CPUs.
so when my wife asks why i need to build yet another box, i will blame you instead.  thanks!

**Comment by edwardsmarkf** [comment]
thanks to **bdarnell's** extraordinary patience, the following worked for me using centos7 on [virtualbox](https://www.virtualbox.org/wiki/VirtualBox):

```
curl -fsSL   https://get.docker.com/   |  sh  ;
systemctl start docker   ;
systemctl status docker  ;
docker run hello-world    ;   ## test!
git clone  https://github.com/cockroachdb/cockroach.git  ;
git  submodule  update  --init   ;
(edit   rocksdb/cgo_flags.go file and remove the -msse4.2 on line eight)
export GOPATH=~/go   ;
export PATH=$PATH:$GOPATH/bin   ;
build/builder.sh  make build    ;
./cockroach  version    ;
export PATH=$PATH:$(pwd) ;   ##
cd ;   # go home
which cockroach   ;
cockroach  version    ;
```

**Comment by knz** [comment]
Tamir I'm reassinging this to you since you have #14466  ready already.

**Comment by edwardsmarkf** [comment]
tip:    **more  /proc/cpuinfo;**  

look under "flags" look to see if sse4_1 and and sse4_2 are there.

**Comment by tamird** [comment]
Fixed (really this time) in #15247.

**Comment by ansarizafar** [comment]
I am still getting same error with the latest version
```
zafar@home:~/Downloads/cockroach$ ./cockroach version
Build Tag:    v1.0-rc.1-dirty
Build Time:   2017/05/01 18:33:34
Distribution: CCL
Platform:     linux amd64
Go Version:   go1.8.1
C Compiler:   gcc 6.3.0
Build SHA-1:  2d4d1ab5c42efb5accf73c9876e6ffd934fdc9e6
Build Type:   release
zafar@home:~/Downloads/cockroach$ ./cockroach start --insecure 
*
* WARNING: [n?] running in insecure mode, this is strongly discouraged. See --insecure.
*
zafar@home:~/Downloads/cockroach$ ./cockroach start --insecure --background
*
* WARNING: [n?] running in insecure mode, this is strongly discouraged. See --insecure.
*
E170502 16:50:01.306436 1 cli/error.go:68  exit status 2
Error: exit status 2
Failed running "start"
zafar@home:~/Downloads/cockroach$ 

```
